docs/tutorial/02-making-improvements.md

```diff
@@ -0,0 +1,269 @@
+
+## Step 1: 🔒 Create a Private Helper Function
+
+First, let's extract the number generation into a private helper function. This follows the DRY principle (Don't Repeat Yourself) and makes our code more maintainable.
+
+Open `contracts/guess-the-number/src/lib.rs` and add this private function inside the `impl GuessTheNumber` block:
+
+```rust
+#[contractimpl]
+impl GuessTheNumber {
+    // ... existing functions ...
+
+    /// Private helper function to generate and store a new random number
+    fn set_random_number(env: &Env) {
+        let new_number: u64 = env.prng().gen_range(1..=10);
+        env.storage().instance().set(&THE_NUMBER, &new_number);
+    }
+}
+```
+
+### Understanding Private Functions
+
+Notice that this function doesn't have `pub` in front of it - this makes it private. Private functions:
+
+- Can only be called from within the same contract
+- Don't become part of the contract's public API
+- Are useful for internal logic and code reuse
+- Help keep your contract interface clean and focused
+
+## Step 2: 👷‍♂️ Update the Constructor
+
+Now let's modify the `__constructor` to set an initial number when the contract is deployed:
+
+```rust
+pub fn __constructor(env: &Env, admin: &Address) {
+    Self::set_admin(env, admin);
+    Self::set_random_number(env); // Add this line
+}
+```
+
+### Why This Improves Things
+
+By setting a number in the constructor:
+
+1. **Immediate functionality**: The contract works right after deployment
+2. **No crash risk**: `guess` will never encounter a missing number
+3. **Better user experience**: Players can start guessing immediately
+
+## Step 3: ♻️ Update the Reset Function
+
+Let's simplify our `reset` function to use the new helper:
+
+```rust
+/// Update the number. Only callable by admin.
+pub fn reset(env: &Env) {
+    Self::require_admin(env);
+    Self::set_random_number(env);
+}
+```
+
+Much cleaner! The logic is now centralized in our helper function. Note that this is still a public function, see the `pub`? The distinction between "public" and "private" might seem confusing here. Let's run the application and it should clear everything up:
+
+```bash
+$ npm start
+```
+
+In the header menu, you should see a `</> Debugger` link. This will open up our handy dandy Contract Explorer. This will let you explore all the contracts within your project, view the documentation for every method, and even invoke them with arguments directly from the UI!
+
+This is a great tool to help debug and test while you develop. So give it a shot! Navigate to the debugger, select the `guess_the_number` contract, and test out our new `reset` method by hitting it's `Submit` button.
+
+Our `reset` method is available to be called by code _outside_ our contract because we opted in to it being a public method with the `pub`. Our `set_random_number` is private by default, it's not even visible to the outside world. It's not listed in the Contract Explorer. It's not listed in the CLI help either:
```

Comment:
```suggestion
Our `reset` method is available to be called by code _outside_ our contract because we opted in to it being a public method with the `pub` keyword. Our `set_random_number` is private by default, it's not visible to the outside world. It's not listed in the Contract Explorer. It's not listed in the CLI help either:
```